mantidproject/mantid
# Quiet down DirectILLDiagnostics

**Description of work.**

This PR replaces the `*=` operator used in `DirectILLDiagnostics` by calls to `CreateSingleValuedWorkspace` and `Multiply` to give better control over logging.

**Report to:** [nobody].

**To test:**

With unit test data directory in data search directories, run the following script:
```python
ws = DirectILLCollectData('ILL/IN5/104007.nxs')
diag = DirectILLDiagnostics(ws)
```

Check that the call to `DirectILLDiagnostics` produces only the log below when logging level is set to *Notice*:
```
DirectILLDiagnostics started
Default mask loaded from IN5_Mask.xml
Background diagnostics disable by the IPF.
Elastic peak diagnostics disabled by the IPF.
DirectILLDiagnostics successful, Duration 4.96 seconds
```


Fixes #23597.

*This does not require release notes* because **this does not contain any functional changes.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
